lib/node_modules/@stdlib/array/base/mskfilter-map/docs/types/test.ts

```diff
@@ -0,0 +1,102 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import mskfilterMap = require( './index' );
+
+function clbk( value: number ): number {
+	return value;
+}
+
+// TESTS //
+
+// The function returns an array...
+{
+	mskfilterMap( [ 1, 2, 3, 4 ], [ 0, 0 ], clbk ); // $ExpectType number[]
+	mskfilterMap( [ 1, 2, 3, 4 ], [ 0, 0 ], clbk, {} ); // $ExpectType number[]
+	mskfilterMap<any>( [ 1, 2, 3, 4 ], [ 0, 0 ], clbk ); // $ExpectType any[]
+	mskfilterMap<any>( [ 1, 2, 3, 4 ], [ 0, 0 ], clbk, {} ); // $ExpectType any[]
+	mskfilterMap<number>( [ 1, 2, 3, 4 ], [ 0, 0 ], clbk ); // $ExpectType number[]
+	mskfilterMap<number>( [ 1, 2, 3, 4 ], [ 0, 0 ], clbk, {} ); // $ExpectType number[]
+	mskfilterMap<string>( [ '1', '2', '3', '4' ], [ 0, 0 ], clbk ); // $ExpectType string[]
```

Comment:
This shouldn't be valid. We've lost type information. The `clbk` is typed to accept numbers, but the elements which are passed to it will be strings. The `index.d.ts` file needs to be updated accordingly.